Барнаева М.К. Лаб. 6 Группа 6311

Barnaeva opened the pull request:
**ФИО:** Барнаева Марина
**Номер группы:** 6311
**Номер лабораторной:** 6
**Краткое описание исходного кода:** Я взяла исходный код из своей старой лабораторной работы. Вот ссылка на него: https://github.com/Barnaeva/isb_barnaeva/blob/lab1-var3/lab1-var3/TASK2/task2.py.
**Краткое описание проделанной работы:** Я сделала юнит-тесты.

IlyaOv commented on lab_3/config.py:
```diff
@@ -0,0 +1,16 @@
+    "blood_type": "^(A|B|AB|O)[+−]$",
+    "issn": "^\\d{4}-\\d{4}$",
+    "uuid": "^[a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12}$",
+    "date": "^\\d{4}-\\d{2}-\\d{2}$",
```
Для даты предлагаю уточнить выражение.

IlyaOv reviewed (approved):
Зачтено

Barnaeva committed:
test

Barnaeva committed:
Fixed import order to comply with PEP 8 standards

Barnaeva committed:
Fixed E501 (too long line) according to PEP 8 recommendations.

Barnaeva committed:
F401: Unused imports removed
E302: Added blank lines between functions
E251: Fixed spacing around parameters

Barnaeva committed:
Fixed E128, E129, E131, E722 in accordance with PEP 8

Barnaeva committed:
Fixed F841 in accordance with PEP 8

Barnaeva committed:
Fixed N806,E123 in accordance with PEP 8

Barnaeva committed:
N802, N803, N806 - Variable/function naming
E401, E402 - Import order and grouping

Barnaeva committed:
created a module for tests from two taken from the old repository

Barnaeva committed:
first test

Barnaeva committed:
tests for make_key and decryption_cod3

Barnaeva committed:
test for encryption and decryption

Barnaeva committed:
added parameterized test

Barnaeva committed:
added a test to check for exceptions.

IlyaOv commented on lab6/unit_tests.py:
Общие замечания:
1. Давайте сделаем еще 1 менее примитивный тест, как в задании.
2. Код желательно привести к PEP 8 по части импортов, пустых строк и т.д.
3. ЛР должна выполняться в lab_6, чтобы запустился action, который проверяет тесты.

Barnaeva committed:
fixes

Barnaeva committed:
+ requirements.txt

Barnaeva committed:
rename module test_unit.py

IlyaOv reviewed (approved):
Зачтено
